consensus/src/block_storage/block_tree.rs

```diff
@@ -285,12 +285,19 @@ impl BlockTree {
             .or_insert_with(|| Arc::clone(&qc));
 
         if self.highest_ordered_cert.commit_info().round() < qc.commit_info().round() {
-            self.highest_ordered_cert = qc;
+            // Question: We are updating highest_ordered_cert but not highest_ordered_root. Is that fine?
```

Comment:
the root is updated by the send_for_execution I think

Reply:
Is there a need to distinguish between highest_ordered_cert and highest_ordered_root? 
Can't we always derive the highest_ordered_root from highest_ordered_cert?

Reply:
I don't remember, but probably not change in this big pr